### PR TITLE
Allow Symfony 4

### PR DESCRIPTION
**Pull request by garak**
See #3124

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -82,6 +82,11 @@ jobs:
 
         -
             <<: *STANDARD_TEST_JOB
+            php: 7.1
+            env: PHP_CS_FIXER_TEST_USE_LEGACY_TOKENIZER=1 SYMFONY_VERSION="^4.0@dev"
```
Comment (keradus) [inline .travis.yml]: Legacy tokenizer has to be used in at least one job, but no need to have it multiple times - we want to test it, but it makes job slower.
Also, there is no need to have new job to handle Symfony4, we could reuse existing job.
Please combine both 7.1 jobs into one (so, basically, drop previous job)

**Comment by keradus** [comment]
Please rebase

**Comment by garak** [comment]
@keradus if I replace `ProcessBuilder` with `Process`, how can I apply prefix?

**Comment by keradus** [comment]
```diff
-ProcessBuilder::create(['-l']->setPrefix('ls')->getProcess()
+new Process('ls -l')
```

File: composer.json [hunk 2]
```diff
@@ -21,24 +21,24 @@
...
+        "symfony/process": "^2.3 || ^3.0 || ^4.0",
+        "symfony/stopwatch": "^2.5 || ^3.0 || ^4.0"
     },
     "require-dev": {
         "johnkary/phpunit-speedtrap": "^1.0.1",
         "justinrainbow/json-schema": "^5.0",
         "phpunit/phpunit": "^4.8.35 || ^5.4.3",
-        "php-coveralls/php-coveralls": "^1.0.2",
-        "symfony/phpunit-bridge": "^3.2.2"
+        "satooshi/php-coveralls": "^1.0.2@dev",
```
Comment (keradus) [inline composer.json]: please drop `@dev`. 1.0.2 is `@stable`.

Comment (keradus) [reply composer.json]: even more, sth wrong with your rebase.
it shall be `"php-coveralls/php-coveralls"`, not `"satooshi/php-coveralls"`

File: src/Linter/ProcessLinterProcessBuilder.php [hunk 3]
```diff
@@ -50,6 +49,8 @@ public function build($path)
             array_unshift($arguments, '--php');
         }
 
-        return ProcessBuilder::create($arguments)->setPrefix($this->executable)->getProcess();
+        $process = new Process('\''.$this->executable.'\' \''.implode('\' \'', $arguments).'\'');
```
Comment (keradus) [inline src/Linter/ProcessLinterProcessBuilder.php]: extra escaping and escaping and concatenation and again and again is so unreadable...

**Comment by keradus** [comment]
> garak reviewed 11 hours ago

wow, you are reviewing your own PR :D 

> It looks like everything is fine now

actually, not, look at statuses

**Comment by garak** [comment]
Well, I don't know what to do with appveyor, it's obviously related to ProcessBuilder/Process change, where I just tried a wild guess.
I'm open to suggestions

**Comment by keradus** [comment]
I already commented the line where you do some weird ocncat/escape that is causing the issue.
Just read the error from CI service and that's it... https://ci.appveyor.com/project/keradus/php-cs-fixer/build/dev-7925/job/jdtlrxumqo3hb6as#L253

**Comment by garak** [comment]
Question is you provided me a partial hint:
```diff
-ProcessBuilder::create(['-l']->setPrefix('ls')->getProcess()
+new Process('ls -l')
```
But there's no simply `['-l']`, there's an array with many elements.
From what I understand in tests, elements in array can be quoted or unquoted, but I still don't see when the quotes are needed and when are not.

**Comment by keradus** [comment]
quoting is required for CLI params that could could contain space, and is optional for others.

**Comment by garak** [comment]
Is that something that we need to take care? Or it's supposed to be in charge of Process component?
I'm sorry but docs for 4.0 are still not up to date, so I don't know when to find such informations.
I also found a (maybe) related issue, that is still open https://github.com/symfony/symfony/issues/20527#issuecomment-318663546

**Comment by keradus** [comment]
yes, it's sth that need to be taken care here.
4.0 is just a deprecations drop, documentation for 3.x is valid for 4.0 without deprecations.

**Comment by keradus** [comment]
cool. then you aware aware of deprecations and could ignore deprecated part on your own.
I don't see not updated documentation anyhow connected to this PR.

**Comment by garak** [comment]
Here is the non-updated doc: http://symfony.com/doc/master/components/process.html#executing-php-code-in-isolation (including ProcessBuilder, even if it's not in master/4.0 anymore)
So, what's next? Should I revert the removal of ProcessBuilder?

**Comment by keradus** [comment]
again, not-up-to-date symfony docs is not our concern.

> So, what's next?

fix issues raised by CIs

> Should I revert the removal of ProcessBuilder?

then you will again have crashing build on trying to use unexisting class

**Comment by garak** [comment]
So I don't get you when you say
> you are aware of deprecations and could ignore deprecated part on your own

How can I ignore deprecated part?

**Comment by keradus** [comment]
if official Symfony 4 docs has paragraph about code that is deprecated and you know it, skip reading this paragraph.
or jump into Symfony 4 codebase repo directly. 

man, you going in circle. you know ProcessBuilder is deprecated, I raised this in comment on very beginning of this repo. yes, to allow PHP CS Fixer you need to stop using it and replace it with `Process`. If you don't know how `Process` works open documentation of last 3.x, of 4.0, github repo directly, doesn't matter.

**Comment by keradus** [comment]
Thank you @garak.

**Pull request merged**